### PR TITLE
style: improve docs links, toc and icons

### PR DESCRIPTION
**Pull request by alexiscolin**
Last polish docs website design by:

- Limiting ToC height and allowing scroll inside (cf #1513)
- Replacing external icon
- Adding the full logo
- Improving the caret icon
- Fixing Toc hover color on dark mode

### CODE REVIEW

**Review by leohhhn — APPROVED** [review]
Looks good! Thanks.

File: misc/docusaurus/docusaurus.config.js [hunk 1]
```diff
@@ -45,11 +45,10 @@ const config = {
     ({
       navbar: {
         hideOnScroll: true,
-        title: "Gno.land",
```
[the remark this answers is not in the document]
Comment (alexiscolin) [reply misc/docusaurus/docusaurus.config.js:48]: Because we want the logo in the navbar instead of a plain text.

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

Please make sure there is no funky business when running `yarn build` for the site 🙏

**Comment by alexiscolin** [comment]
@zivkovicmilos The `yarn build` cmd built the website in success 🙂

Capture d’écran 2024-01-10 à 19 25 19

**Review by thehowl — APPROVED** [review]

**Pull request merged**